src/classed_table_generator.rs

```diff
@@ -0,0 +1,189 @@
+use std::fmt::Write;
+use syntect::{
+    escape::Escape,
+    html::ClassStyle,
+    parsing::{
+        BasicScopeStackOp, ParseState, Scope, ScopeStack, ScopeStackOp, SyntaxReference, SyntaxSet,
+        SCOPE_REPO,
+    },
+    util::LinesWithEndings,
+};
+
+/// The ClassedTableGenerator generates HTML tables of the following form:
+/// <table>
+///   <tbody>
+///     <tr>
+///       <td class="line" data-line="1">
+///       <td class="code">
+///         <span class="hl-source hl-go">
+///           <span class="hl-keyword hl-control hl-go">package</span>
+///           main
+///         </span>
+///       </td>
+///     </tr>
+///   </tbody>
+/// </table
+///
+/// If max_line_len is not None, any lines with length greater than the
+/// provided number will not be highlighted.
+pub struct ClassedTableGenerator<'a> {
```

Comment:
Note that this implementation looks quite long, but most of it is quite simple, and will replace ~200 lines of very complex go code that currently parses the results as HTML and reforms it into this table format.

Reply:
One reason I chose to write this logic in Go originally was because we don't have anyone else at Sourcegraph who is able to dive into this code-base and work on it if we need to. My time these days is also particularly limited, so if there are changes needed to this code in the future it will likely be a "camden or.. we may not be able to do it any time soon"

I don't think that is a blocker here, but thought it worth raising as a point.

Also, let's make sure to add tests for this code before merging.